### PR TITLE
Cherry-picking WorkspaceTestUtil fix from 7.30.x to main

### PR DESCRIPTION
**Pull request by ScrewTSW**
Signed-off-by: Tibor Dancs <tdancs@redhat.com>

### What does this PR do?
Fixes an edge case when the stop workspace request would return 409 if the workspace is already stopped and cause a test failure

### What issues does this PR fix or reference?

cherry-pick of #19993 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by Ohrimenko1988 — APPROVED** [review]

File: tests/e2e/utils/workspace/TestWorkspaceUtil.ts [hunk 1]
```diff
@@ -164,31 +164,41 @@ export class TestWorkspaceUtil implements ITestWorkspaceUtil {
...
+        }
+        
+        // response code 204: "No Content" expected
+        if (stopWorkspaceResponse.status !== 204) {
+            throw new Error(`Can not stop workspace. Code: ${stopWorkspaceResponse.status} Data: ${stopWorkspaceResponse.data}`);
+        }
+
+        let stopped: boolean = false;
+        let wsStatus = await this.processRequestHandler.get(stopWorkspaceApiUrl);
+        for (let i = 0; i < TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS; i++) {
```
Comment (Katka92) [inline tests/e2e/utils/workspace/TestWorkspaceUtil.ts:189]: It's weird to see `TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS` here but it doesn't have to be fixed in this PR.

**Review by Katka92 — APPROVED** [review]

File: tests/e2e/utils/workspace/TestWorkspaceUtil.ts (hunk 1 above)
Comment (ScrewTSW) [reply tests/e2e/utils/workspace/TestWorkspaceUtil.ts:189]: yeah, and there's also a typo in the variable name :D
not sure what I should put there instead
